eventsocket/eventsocket.go

```diff
@@ -45,6 +49,63 @@ type Connection struct {
 	textreader    *textproto.Reader
 	err           chan error
 	cmd, api, evt chan *Event
+	hub           *RequestHub
+}
+
+// Request is a bgapi request struct
+type Request struct {
+	UUID      string
+	timestamp time.Time
+	resp      chan *Event
+	err       chan error
+	command   string
+}
+
+// RequestHub is the asyn request/reply hub for bgapi commands
+type RequestHub struct {
+	requests  map[string]*Request
+	inbound   chan *Request
+	responses chan *Event
+	timeout   chan bool
+}
+
+func (h *RequestHub) run() {
+	go func() {
```

Comment:
Any particular reason to do the timeout this way instead of adding a 

`case <-time.After(time.Second` to the select statement below?